just_audio/android/src/main/java/com/ryanheise/just_audio/AudioPlayer.java

```diff
@@ -576,52 +580,63 @@ private MediaSource getAudioSource(final Object json) {
-        switch ((String)map.get("type")) {
-        case "progressive":
-            return new ProgressiveMediaSource.Factory(buildDataSourceFactory())
-                    .createMediaSource(new MediaItem.Builder()
-                            .setUri(Uri.parse((String)map.get("uri")))
-                            .setTag(id)
-                            .build());
-        case "dash":
-            return new DashMediaSource.Factory(buildDataSourceFactory())
-                    .createMediaSource(new MediaItem.Builder()
-                            .setUri(Uri.parse((String)map.get("uri")))
-                            .setMimeType(MimeTypes.APPLICATION_MPD)
-                            .setTag(id)
-                            .build());
-        case "hls":
-            return new HlsMediaSource.Factory(buildDataSourceFactory())
-                    .createMediaSource(new MediaItem.Builder()
-                            .setUri(Uri.parse((String)map.get("uri")))
-                            .setMimeType(MimeTypes.APPLICATION_M3U8)
-                            .build());
-        case "silence":
-            return new SilenceMediaSource.Factory()
-                    .setDurationUs(getLong(map.get("duration")))
-                    .setTag(id)
-                    .createMediaSource();
-        case "concatenating":
-            MediaSource[] mediaSources = getAudioSourcesArray(map.get("children"));
-            return new ConcatenatingMediaSource(
-                    false, // isAtomic
-                    (Boolean)map.get("useLazyPreparation"),
-                    decodeShuffleOrder(mapGet(map, "shuffleOrder")),
-                    mediaSources);
-        case "clipping":
-            Long start = getLong(map.get("start"));
-            Long end = getLong(map.get("end"));
-            return new ClippingMediaSource(getAudioSource(map.get("child")),
-                    start != null ? start : 0,
-                    end != null ? end : C.TIME_END_OF_SOURCE);
-        case "looping":
-            Integer count = (Integer)map.get("count");
-            MediaSource looperChild = getAudioSource(map.get("child"));
-            return new LoopingMediaSource(looperChild, count);
-        default:
-            throw new IllegalArgumentException("Unknown AudioSource type: " + map.get("type"));
+        Map<?, ?> map = (Map<?, ?>) json;
+        String id = (String) map.get("id");
+
+        DrmSessionManager drmSessionManager = null;
+        if (map.get("drm") != null) {
+            Map<?, ?> drm = (Map<?, ?>) map.get("drm");
+            if (drm.get("clearKey") != null) {
+                drmSessionManager = new DefaultDrmSessionManager.Builder()
+                        .setUuidAndExoMediaDrmProvider(C.CLEARKEY_UUID, FrameworkMediaDrm.DEFAULT_PROVIDER).
+                                build(new LocalMediaDrmCallback(((String) drm.get("clearKey")).getBytes()));
+            }
+        }
+
+        switch ((String) map.get("type")) {
```

Comment:
I'm not sure if I want to change the project's formatting style, but if we do it, it should be first proposed in an issue and done as a separate pull request, otherwise it adds changes to the diff that are harder to review.

Reply:
Changes not related to formatting but setDrmSessionManager(drmSessionManager) added to the various MediaSource factories.

Reply:
Would you be able to undo the formatting changes to this file?